.circleci/config.yml

```diff
@@ -0,0 +1,138 @@
+            sudo apt-get install xsltproc
+            ./build-support/go-offline.sh
+
+      - save_cache:
+          key: maven-dependencies-{{ checksum "pom.xml" }}
+          paths:
+            - ~/.m2
+
+      - run:
+          name: Tests
+          command: ./mvnw clean verify
+
+      - run:
+          name: Collect test reports
+          when: always
+          command: |
+            mkdir -p /tmp/test-reports/junit/
+            find . -type f -regex ".*/target/surefire-reports/.*xml" -exec cp {} /tmp/test-reports/junit/ \;
+
+      - store_test_results:
+          path: /tmp/test-reports/
+
+  publish_snapshot:
+    docker:
+      - image: circleci/openjdk:8-jdk
+    steps:
+      - checkout
+      - restore_cache:
+          key: maven-dependencies-{{ checksum "pom.xml" }}
+      - run:
+          name: Publish snapshot
+          command: |
+            # Guard against running this on pull requests or forks
+            [ "$CIRCLE_PROJECT_USERNAME" == "openzipkin" ] || exit 0
+            [ -z "$CIRCLE_PR_NUMBER" ] || exit 0
+            ./build-support/publish-snapshot.sh
+
+  publish_stable:
+    docker:
+      - image: circleci/openjdk:8-jdk
+    steps:
+      - checkout
+      - restore_cache:
+          key: maven-dependencies-{{ checksum "pom.xml" }}
+      - run:
+          name: Publish stable version
+          command: |
+            # Guard against running this on pull requests or forks
+            [ "$CIRCLE_PROJECT_USERNAME" == "openzipkin" ] || exit 0
+            [ -z "$CIRCLE_PR_NUMBER" ] || exit 0
+            ./build-support/publish-stable.sh
+
+  trigger_publish:
+    docker:
+      - image: circleci/openjdk:8-jdk
+    steps:
+      - checkout
+      - restore_cache:
+          key: maven-dependencies-{{ checksum "pom.xml" }}
+      - run:
+          name: Configure Git for release
+          command: |
+            # parameters used during release
+            # allocate commits to CI, not the owner of the deploy key
+            git config user.name "zipkinci"
+            git config user.email "zipkinci+zipkin-dev@googlegroups.com"
+            # setup https authentication credentials, used by ./mvnw release:prepare
+            git config credential.helper "store --file=.git/credentials"
+            echo "https://$GH_TOKEN:@github.com" > .git/credentials
+      - run:
+          name: Trigger publish
+          command: |
+            # Guard against running this on pull requests or forks
+            [ "$CIRCLE_PROJECT_USERNAME" == "openzipkin" ] || exit 0
+            [ -z "$CIRCLE_PR_NUMBER" ] || exit 0
+            ./build-support/trigger-publish.sh
+
+workflows:
+  version: 2
+  build_and_publish:
+    jobs:
+      - build
+      - publish_snapshot:
+          requires:
+            - build
+          filters:
+            branches:
+              only: master
+      - publish_stable:
+          requires:
+            - build
+          filters:
+            branches:
+              ignore: /.*/
```

Comment:
Without this, the job will run on all branches. See [this section of the documentation](https://circleci.com/docs/2.0/workflows/#git-tag-job-execution):

> 1. For a branch push unaffected by any filters, CircleCI runs the job.
> 2. For a tag push unaffected by any filters, CircleCI skips the job.